docs/mobile.md

```diff
@@ -0,0 +1,115 @@
+`chaininterface::FeeEstimator` trait and thus servicing smoothly the rest of the LDK engine. This
+utility will be a RPC HTTP client, configurable to be pointed toward any Bitcoin Core full-node.
+
+## Liveliness & Watchtowers
+
+A Lightning node must always monitor blockchain updates to react in consequence, either in case of
+a revoked commitment transaction confirmed or reaching a given block height expirying a HTLC. This
+is a strong requirement as even missing few hours of blocks might trigger a loss of funds. Mobile
+devices with an unreliable access to Internet or often shutdown by their users are particularly
+sensible to this issue.
+
+This liveliness requirement is function of which Lightning situation is considered. 
+
+As soon, as a payment has been initiated on the channel, your counterparty can broadcast _any old  
+commitment_ at _any time_ thus forcing your Lightning node to be always online in prevision of such
+an event. Watchtowers have been designed by the Lightning community to delegate such justice
+enforcement function to an external entity. As this entity is trusted to broadcast client penalty
+transactions, it is recommended to rely on multiple of them.
+
+Currently, the LDK community is engaged on adoption of BOLT13, a cross-implementation standard
+for watchtowers. LDK hope to offer soon a client-side implementation of BOLT13, pluggable on
+the LDK engine. Beyond, if channels are opened with low-trust counterparties, it's recommended
+to use longuer timelocks delays in your configuration (`ChannelHandshakeConfig::our_to_self_delay`).
+
+With regards to handling in-flight HTLCs, a Lightning node not online when a HTLC has expired is
+taking the risk of seeing its counterparty canceling onchain the HTLC and thus loosing funds. 
+
+In case of sending a payment, if it doesn't succeed immediately (e.g waiting on the receiver to
+be online), the mobile should be back online later to remove the sent HTLC in cooperation with
+channel counterparty or worst-case scenario to broadcast onchain the commitment transaction and
+a corresponding HTLC-timeout. Setting a long CLTV delay at first hop relay offers a _time window_
+for how long the mobile device can be offline. Ideally, the Lightning application will send a
+notification to the user at half of the CLTV delay warning about the requirement to connect back
+to the network.
+
+On the UX-side, as soon the HTLC is sent on the channel, the user balance should be decreased
+accordingly as unlock secret might already be known by counterparty or receiver. In the future,
+the PTLC protocol extension will allow to acknowledge that the payment has been received by 
+expected payee before to decrease balance.
+
+In case of receiving a payment, if it doesn't succeed immediately (e.g waiting on payer interactions
+to receive the preimage), the mobile should be back online to hopefully learn payment preimage
```

Comment:
Hmm? I mean if we're the recipient we either know the preimage or we don't, there's not really a learn-it-later outcome unless we're forwarding.

Reply:
I think you can have more complex settlement scenarios. E.g when the HTLC is used for a swap or as a fidelity bond.